src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -418,7 +418,8 @@ type WarningCode =
     | QirEmissionGeneratedWarning = 8002
 
     | InvalidAssemblyProperties = 8101
-    | MissingTargetInstructionName = 8102
+    | DuplicateAssemblyProperty = 8102
```

Comment:
Can we change the number of existing error codes?

Reply:
Nothing should be relying on the numbering and everything should use the DU case instead. 
At some point we should ideally revise the setup to allow for conversions between warnings and errors, and to facilitate documenting error codes online. I am not sure if we should try to keep the numbers stable before that but let's give it a try and that will help inform whether we think we can do that for 1.0 (and once we have done these changes).
I changed the numbering to add the new diagnostic as 8103.